src/olympia/scanners/tests/test_admin.py

```diff
@@ -601,3 +609,214 @@ def test_get_fields_for_non_admins(self):
+        assert response.redirect_chain == [(self.list_url, 302)]
+        assert run_yara_query_rule_mock.call_count == 1
+        assert run_yara_query_rule_mock.call_args[0] == (rule.pk,)
+        messages = list(response.context['messages'])
+        assert len(messages) == 1
+        assert 'successfully queued' in str(messages[0])
+        rule.reload()
+        assert rule.state == RUNNING
+
+    @mock.patch('olympia.scanners.admin.run_yara_query_rule.delay')
+    def test_run_action_wrong_state(self, run_yara_query_rule_mock):
+        rule = ScannerQueryRule.objects.create(
+            name='bar', scanner=YARA, state=ABORTING)
+        response = self.client.post(
+            reverse(
+                'admin:scanners_scannerqueryrule_handle_run',
+                args=[rule.pk],
+            ), follow=True
+        )
+        assert response.status_code == 200
+        assert response.redirect_chain == [(self.list_url, 302)]
+        assert run_yara_query_rule_mock.call_count == 0
+        messages = list(response.context['messages'])
+        assert len(messages) == 1
+        assert 'could not be queued' in str(messages[0])
+        rule.reload()
+        assert rule.state == ABORTING
+
+    def test_run_action_no_permission(self):
+        user = user_factory()
+        self.grant_permission(user, 'Admin:Curation')
+        self.client.login(email=user.email)
+        rule = ScannerQueryRule.objects.create(
+            name='bar', scanner=YARA, state=NEW)
+        response = self.client.post(
+            reverse(
+                'admin:scanners_scannerqueryrule_handle_run',
+                args=[rule.pk],
+            ), follow=True
+        )
+        assert response.status_code == 404
+
+    def test_abort_action(self):
+        rule = ScannerQueryRule.objects.create(
+            name='bar', scanner=YARA, state=RUNNING)
+        response = self.client.post(
+            reverse(
+                'admin:scanners_scannerqueryrule_handle_abort',
+                args=[rule.pk],
+            ), follow=True
+        )
+        assert response.status_code == 200
+        assert response.redirect_chain == [(self.list_url, 302)]
+        messages = list(response.context['messages'])
+        assert len(messages) == 1
+        assert 'being aborted' in str(messages[0])
+        rule.reload()
+        assert rule.state == ABORTING
+
+    def test_abort_action_wrong_state(self):
+        rule = ScannerQueryRule.objects.create(
+            name='bar', scanner=YARA, state=COMPLETED)
+        response = self.client.post(
+            reverse(
+                'admin:scanners_scannerqueryrule_handle_abort',
+                args=[rule.pk],
+            ), follow=True
+        )
+        assert response.status_code == 200
+        assert response.redirect_chain == [(self.list_url, 302)]
+        messages = list(response.context['messages'])
+        assert len(messages) == 1
+        assert 'could not be aborted' in str(messages[0])
```

Comment:
Should you also assert that the rule's `pk` and `state` appear in the message?

Reply:
As above.